### DISCUSSION TITLE
Quickstart React Profile Image not correctly running updateProfile function

### DISCUSSION THREAD

**Discussion by c-dot-gonz** [supabase/supabase#12061] [Questions]
Hello! I'm new to both React and Supabase so I'm going through the [Quickstart tutorial](https://supabase.com/docs/guides/getting-started/tutorials/with-react#add-the-new-widget). I've noticed that once the widget is added, if the Avatar is uploaded, it won't actually run the updateProfile function created earlier in the tutorial unless you click the "Update Profile" form button, even though running updateProfile is part of the Avatar widget, but it still uploads the image file to the storage bucket. Here's the code in question:

`                <form onSubmit={updateProfile} className="form-widget">
                    <Avatar
                    url={avatar_url}
                    size={150}
                    onUpload={(url) => {
                        setAvatarUrl(url)
                        updateProfile({ username, website, avatar_url: url })
                    }}
                    />
                    <div>Email: {session.user.email}</div>`

Those first and last lines are just context for where I placed the widget in relation to other stuff from the tutorial. While I initially typed all the code out, I copied and pasted the tutorial code to test and I still get the same behavior.

Testing steps:
On the Account page log in then click Upload an Avatar
Select an image to upload and click Open
The current image should be replaced with the new image
Once you see the new image, refresh the page
The old avatar image comes back.

Is this the intended behavior of the script?

**Comment by GaryAustin1** [comment]
Your image is probably cached in either the browser or CDN.  Default is 1 hour.
https://supabase.com/docs/guides/storage/cdn#cache-duration

**Reply by c-dot-gonz** [reply]
Interesting. Good to know that the default is 1 hour. However, I refreshed the page after an hour and did a cache refresh to clear the browser side and not only does the page still show the avatar, the "profiles" table shows the original avatar_url instead of any of the updated ones, even though they appear in the "avatars" bucket with newer timestamps. Is there anything else I should be trying?
